transports/quic/src/transport.rs

```diff
@@ -223,7 +286,38 @@ impl<P: Provider> Transport for GenTransport<P> {
         }
 
         if let Poll::Ready(Some(ev)) = self.listeners.poll_next_unpin(cx) {
-            return Poll::Ready(ev);
+            match ev {
+                TransportEvent::Incoming {
+                    listener_id,
+                    upgrade,
+                    local_addr,
+                    send_back_addr,
+                } => {
+                    let socket_addr =
+                        multiaddr_to_socketaddr(&send_back_addr, self.support_draft_29)
+                            .unwrap()
+                            .0;
+
+                    if let Some(sender) = self.hole_punch_attempts.remove(&socket_addr) {
+                        if let Err(upgrade) = sender.send(upgrade) {
+                            return Poll::Ready(TransportEvent::Incoming {
+                                listener_id,
+                                upgrade,
+                                local_addr,
+                                send_back_addr,
+                            });
+                        }
+                    } else {
+                        return Poll::Ready(TransportEvent::Incoming {
+                            listener_id,
+                            upgrade,
+                            local_addr,
+                            send_back_addr,
+                        });
```

Comment:
If we don't use `else`, we can unify those two branches right?

Reply:
I didn't get what you mean by not using `else`. Can you suggest the code change?